Assets/Scripts/UI/InGame/Graphics/GraphicsController.cs

```diff
@@ -11,91 +9,166 @@ public class GraphicsController : MonoBehaviour {
+			}
+			catch(Exception ex)
+			{
+				Debug.LogError(ex.ToString());
+				label.color = Color.red;
+				label.text = "error saving player prefs";
+			}
+			// graphics
+			
 		}
+
+		// NULL REFERENCE HAPPENES HERE
 		public void LoadGraphicPlayerPrefs()
 		{
-			
-			var loaded = JsonUtility.FromJson<GeneralGraphics.Data>(PlayerPrefs.GetString("GraphicsData"));
+			try
+			{
+				var loaded1 = JsonUtility.FromJson<GeneralGraphics.GraphicsData>(PlayerPrefs.GetString("GraphicsData"));
+				var loaded2 = JsonUtility.FromJson<QualitySwitcher.QualityData>(PlayerPrefs.GetString("QualityProfile"));
+				var loaded3 = JsonUtility.FromJson<FPSLimiter.FPSData>(PlayerPrefs.GetString("FPSLimit"));
+
+				QualitySwitcher.Slider.value = loaded2.Slider;
+
+				FPSLimiter.FPSDropdown.value = loaded3.dropdown;
+				FPSLimiter.FPSLimit.text = loaded3.field;
+				FPSLimiter.SetFPSLimit();
+
 				
-			GeneralGraphic.TextureQuality.value = loaded.TextureQuality;
-			GeneralGraphic.ShadowRes.value = loaded.ShadowRes;
-			GeneralGraphic.AntiAliasing.value = loaded.AntiAliasing;
-			GeneralGraphic.Shadows.value = loaded.Shadows;
-			GeneralGraphic.VSync.isOn = loaded.VSync;
-			GeneralGraphic.SoftParticles.isOn = loaded.SoftParticles;
-			GeneralGraphic.CustomSettings.isOn = loaded.CustomSettings;
 
-			GeneralGraphic.UpdateEverything();
+				GeneralGraphic.CustomSettings.isOn = loaded1.CustomSettings;
+				AdvancedOptions();
 
-			var _loaded = JsonUtility.FromJson<QualitySwitcher.Data>(PlayerPrefs.GetString("QualityProfile"));
-			QualitySwitcher.Slider.value = _loaded.Slider;
-			
-			label.color = Color.green;
-			label.text = "loaded player prefs";
+				if (loaded1.CustomSettings)
+				{
+					GeneralGraphic.TextureQuality.value = loaded1.TextureQuality;
+					GeneralGraphic.ShadowRes.value = loaded1.ShadowRes;
+					GeneralGraphic.AntiAliasing.value = loaded1.AntiAliasing;
+					GeneralGraphic.Shadows.value = loaded1.Shadows;
+					GeneralGraphic.VSync.isOn = loaded1.VSync;
+					GeneralGraphic.SoftParticles.isOn = loaded1.SoftParticles;
+					GeneralGraphic.UpdateEverything();
+				}
+
+				GeneralGraphic.UpdateObjects();
+
+				label.color = Color.green;
+				label.text = "loaded player prefs";
+			}
+			catch(Exception ex)
+			{
+				Debug.LogError(ex.ToString());
+
+				label.color = Color.red;
+				label.text = "error loading player prefs";
+			}
 		}
 
 		public void AdvancedOptions()
 		{
-			var selected = CustomSettings.isOn;
-			var temp = QualitySettings.GetQualityLevel();
-			
-			if(!selected)
+			if(CustomSettings.isOn)
+			{
+				QualitySettings.SetQualityLevel(CUSTOM, true);
+				QualitySwitcher.Slider.interactable = false;
+			}
+			else
+			{
+				QualitySettings.SetQualityLevel((int)QualitySwitcher.Slider.value, true);
+				QualitySwitcher.Slider.interactable = true;
+			}
+			GeneralGraphic.SetInteractable(CustomSettings.isOn);
+			GeneralGraphic.UpdateObjects();
+			GeneralGraphic.UpdateEverything();
+		}
+
+		public void DeletePrefs()
```

Comment:
"GraphicsData", "QualityProfile", "FPSLimit" are repeated multiple times throughout the classes. A cleaner approach would be to make these a const.